### PR TITLE
Calibration

### PR DESCRIPTION
**Pull request by PantelisElinas**
Hi @youph and @annitrolla 

I've added code for model calibration together with 2 example notebooks demonstrating calibration for binary and multi-class classification problems.

I feel like this may not be the most elegant implementation but it certainly is functional.

I've placed the new module in stellargraph/core because it doesn't really fit anywhere else. That said, it doesn't fit in core either. Maybe we should create a stellargraph/utils or stellargraph/contrib and move the calibration code to that one. Let me know what you think.

Regards,

P.

### CODE REVIEW

File: demos/calibration/calibration-pubmed-node-classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,3065 @@
+{
```
Comment (annitrolla) [inline demos/calibration/calibration-pubmed-node-classification.ipynb]: <p>The csv dataset is the preprocessed version of the Pubmed data. For public demo this should be either provided in addition, or the preprocessing code made available. </p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (annitrolla) [inline demos/calibration/calibration-pubmed-node-classification.ipynb]: <p>The w-* features of Pubmed are TF/IDF weighted word vectors. From this description it seems like these are binary vectors. I don't see any pre-processing step to make them binary either. </p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (annitrolla) [inline demos/calibration/calibration-pubmed-node-classification.ipynb]: <p>May be to create a cell with this command to show it for the users?</p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (annitrolla) [inline demos/calibration/calibration-pubmed-node-classification.ipynb]: <p>A bit confusing title</p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='111'/>

Comment (annitrolla) [inline demos/calibration/calibration-pubmed-node-classification.ipynb]: <p>May be it makes sense to rouynd 4 digits after comma as then we could see that the results are slightly different</p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='115'/>

File: demos/calibration/calibration-pubmed-link-prediction.ipynb [hunk 2]
```diff
@@ -0,0 +1,1795 @@
+{
```
Comment (annitrolla) [inline demos/calibration/calibration-pubmed-link-prediction.ipynb]: <p>You mention that there is no dropout, but currently it is set to 0.2</p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (annitrolla) [inline demos/calibration/calibration-pubmed-link-prediction.ipynb]: <p>I get Value error here:</p><p>`ValueError: X should be a 1d array`</p><p><br></p><p>and the shapes are:</p><p>((15070, 1), (15070,))</p><p><br></p>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/326/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>

File: demos/calibration/calibration-pubmed-node-classification.ipynb (hunk 1 above)
Comment (PantelisElinas) [reply demos/calibration/calibration-pubmed-node-classification.ipynb]: That's been updated now.

File: demos/calibration/calibration-pubmed-link-prediction.ipynb (hunk 2 above)
Comment (PantelisElinas) [reply demos/calibration/calibration-pubmed-link-prediction.ipynb]: This has been fixed in the library.

File: demos/calibration/calibration-pubmed-node-classification.ipynb (hunk 1 above)
Comment (annitrolla) [reply demos/calibration/calibration-pubmed-node-classification.ipynb]: Before cell 22

Comment (annitrolla) [reply demos/calibration/calibration-pubmed-node-classification.ipynb]: Before cell 30

Comment (annitrolla) [reply demos/calibration/calibration-pubmed-node-classification.ipynb]: Before cell 74

**Comment by PantelisElinas** [comment]
Hi @youph and @annitrolla 

I have updated the implementation based on your suggestions. 

Please have a look and let me know if you require further changes.

Regards,

P.

**Comment by youph** [comment]
Looks good, much clearer walkthrough, thanks!!

**Pull request merged**